kb/prometheus-grafana-setup-cloud.md

```diff
@@ -0,0 +1,252 @@
+                           os.write(response.getBytes(UTF_8));
+                       }
+                   });
+           server.start();
+           return server;
+       } catch (IOException e) {
+           throw new RuntimeException(e);
+       }
+   }
+}
+
+//…
+
+// With your scrape endpoint configured, set the metrics scope in your Workflow service stub and use it to create a Client to start your Workers and Workflow Executions.
+
+//…
+{
+    //Create Workflow service stubs to connect to the Frontend Service.
+    WorkflowServiceStubs service = WorkflowServiceStubs.newServiceStubs(
+               WorkflowServiceStubsOptions.newBuilder()
+                      .setMetricsScope(yourScope()) //set the metrics scope for the WorkflowServiceStubs
+                      .build());
+
+   //Create a Workflow service client, which can be used to start, signal, and query Workflow Executions.
+   WorkflowClient yourClient = WorkflowClient.newInstance(service,
+          WorkflowClientOptions.newBuilder().build());
+}
+
+//…
+
+```
+
+To check whether your scrape endpoints are emitting metrics, run your code and go to [http://localhost:8077/metrics](http://localhost:8077/metrics) to verify that you see all the SDK metrics.
+
+## Prometheus configuration for SDK metrics
+
+For Temporal SDKs, you must have Prometheus installed on your host, and configured to listen on the scrape endpoints exposed in the application code.
+
+For this example, you can run Prometheus locally, or as a docker container. In either case, ensure that you set the listen targets to the ports where you have exposed your scrape endpoints.
+When running Prometheus locally, set your target address to port 8077 that we set in the SDK metrics setup example, in your Prometheus configuration YAML file.
+
+Example:
+
+```
+global:
+ scrape_interval: 10s # Set the scrape interval to every 10 seconds. Default is every 1 minute.
+#...
+
+# set your scrape configuration targets to the ports exposed on your endpoints in the SDK
+scrape_configs:
+ - job_name: 'temporalsdkmetrics'
+   metrics_path: /metrics
+   scheme: http
+   static_configs:
+     - targets:
+       # This is the scrape endpoint where Prometheus listens for SDK metrics.
+       - localhost:8077
+       # You can have multiple targets here, provided they are set up in your application code.
+
+```
+
+See the Prometheus documentation for more details on how you can run Prometheus locally or using docker.
+
+Note that Temporal Cloud exposes metrics through a Prometheus HTTP API endpoint (not a scrape endpoint) that can be configured as a datasource in Grafana. The Prometheus configuration described here is for scraping metrics data on endpoints for SDK metrics only.
+
+To check whether Prometheus is receiving metrics from your SDK target, go to http://localhost:9090 and navigate to Status > Targets. The status of your target endpoint defined in your configuration will show here.
```

Comment:
```suggestion
To check whether Prometheus is receiving metrics from your SDK target, go to http://localhost:9090 and navigate to **Status&nbsp;> Targets**.
The status of your target endpoint defined in your configuration appears here.
```
Formatting for a sequence of UI elements per https://developers.google.com/style/ui-elements#angle-brackets.